content/blogposts/2024/announcing-prompt-library.md

```diff
@@ -0,0 +1,149 @@
+
+![image.png](https://sourcegraph.notion.site/image/https%3A%2F%2Fprod-files-secure.s3.us-west-2.amazonaws.com%2Fe7ce844a-fe2e-4102-b77e-e852aee3841b%2Fb793b037-8544-49a5-beae-5db4b070fff0%2Fimage.png?table=block&id=142a8e11-2658-8085-9d3f-d2abd19566be&spaceId=e7ce844a-fe2e-4102-b77e-e852aee3841b&width=2000&userId=&cache=v2)
+
+![image.png](https://sourcegraph.notion.site/image/https%3A%2F%2Fprod-files-secure.s3.us-west-2.amazonaws.com%2Fe7ce844a-fe2e-4102-b77e-e852aee3841b%2F10c88996-bb30-4aca-b646-167003735728%2Fimage.png?table=block&id=142a8e11-2658-8014-a2a1-cae0d8dc95cd&spaceId=e7ce844a-fe2e-4102-b77e-e852aee3841b&width=2000&userId=&cache=v2)
+
+**Once created:**
+
+You can go to Cody Chat, select your saved prompts, add any necessary details, and run them.
+
+![image.png](https://sourcegraph.notion.site/image/https%3A%2F%2Fprod-files-secure.s3.us-west-2.amazonaws.com%2Fe7ce844a-fe2e-4102-b77e-e852aee3841b%2Fc57ee65b-31cf-40de-841d-c53ceca0fa0d%2Fimage.png?table=block&id=142a8e11-2658-8099-8cf5-d613adaea23b&spaceId=e7ce844a-fe2e-4102-b77e-e852aee3841b&width=2000&userId=&cache=v2)
+
+Alternatively, return to Prompt Library, find your prompt, and copy a permalink to share it with others.
+
+![image.png](https://sourcegraph.notion.site/image/https%3A%2F%2Fprod-files-secure.s3.us-west-2.amazonaws.com%2Fe7ce844a-fe2e-4102-b77e-e852aee3841b%2F2a0e4315-7842-4859-9610-63400d467344%2Fimage.png?table=block&id=142a8e11-2658-803f-9899-ff4ee05310bc&spaceId=e7ce844a-fe2e-4102-b77e-e852aee3841b&width=2000&userId=&cache=v2)
+
+# Example Use Cases
+
+Let’s take a look at three realistic example use cases here.
+
+## 1. “Cody is not giving me code in the right language/framework version”
+
+Imagine you are working with a legacy Java application that uses an older version of the Spring framework. You might find that Cody is generating code snippets using a newer version of Spring, which is incompatible with your project. For instance, your project is using Spring 3.x, but Cody provides examples using Spring Boot 2.x features, which are not available in your current setup.
+
+```java
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@RestController
+public class LegacyService {
+
+    @RequestMapping("/legacy")
+    public String handleRequest() {
+        // This method should use older Spring 3.x annotations and configurations
+        return "Handling request with legacy Spring version";
+    }
+}
+```
+
+To address this, you can create a prompt in the Prompt Library that specifies the framework version you are using. For example:
+
+"Please generate Java code that is compatible with the following specifications:
+
+- Java version: 1.6
+- Spring framework version: 3.x
+- Testing framework: JUnit 4"
+
+By sharing this prompt with your team, everyone can ensure that Cody generates code compatible with your legacy system.
```

Comment:
I would show a gif using this prompt and what else is required to make it work and what it looks like.